lib/protocol/src/openctx-protocol.schema.ts

```diff
@@ -62,11 +69,32 @@ export interface Selector {
      */
     contentContains?: string
 }
-export interface ItemsParams {
+/**
+ * The viewer context sent by client to the provider. 
+ */
+export interface Viewer {
+    /**
+     * The list of respositoy remotes from the workspace.
+     * Eg. ["github.com/sourcegraph/openctx"]
+     */
+    repositoryRemotes?: string[]    
```

Comment:
should we call this codebaseNames instead? The example above has been cleaned up and normalized, while I would expect "remote" to be more raw. From a provider implementor it seems rare to actually care about remote specifics / the client still needs to ensure it doesn't leak secrets from remote URLs.

Reply:
FWIW every time we have introduced an idea of a "repo name" or "codebase" into anything at sourcegraph, it has been imo a mistake. It is not a general concept and breaks when you consider monorepos, forks, weird git checkouts, non-git, etc. 

also typo `repositoy`

Reply:
In that case we can probably skip this altogether. Right now it was required by the Github provider to get context from GH. But another way of doing is to ask user to manually map the workDir to git remote in the provider's settings.

I will skip adding ViewerContext for now.

Reply:
@sqs do you have ideas around a better concept? I assume the idea of linking local state to a "shared state" (ie the repo you are contributing to) is a useful concept. Or is the issue auto-inference of what you are working on?

Reply:
I don't yet. It's really dependent on the specific situation. I mean, something like this is probably fine for now, but I think we should not hardcode it in the protocol spec and instead just pass it as a well-known key in some map.

There are a few ways the `repositoryRemote` would be used, and here are some common pitfalls:
- linking local repo to shared repo - need to ensure it works with forks, maybe should use repo ID (on Sourcegraph or GitHub) to not break when there are renames, some use cases should also use git SHAs to identify state
- understanding the user's current workspace - need to account for monorepo users who don't care about other subprojects or care less, need to account for multi-root workspace users
- other common pitfalls: normalizing git repo URLs is hard, often this breaks for github enterprise server etc, often you shoehorn other naming schemes like packages into a "repo name", often this breaks when the user has no git remotes set up yet (as @ErikaRS saw yesterday in her feedback)